src/phpDocumentor/Compiler/Linker/Linker.php

```diff
@@ -129,6 +132,8 @@ public function substitute($item, ?DescriptorAbstract $container = null)
         }
 
         if (is_iterable($item)) {
+            assert(is_array($item) || $item instanceof ArrayAccess);
```

Comment:
substituteChildrenOfCollection has a type hint iterable. However, in the code of the method, there is something like $item['somethin'] = $foo; which is only possible on arrays or ArrayAccess implementations. At first, I thought this was array only but it's not. I had to document the ArrayAccess part to allow said syntax, but then, Phpstan complained that I was not pushing an ArrayAccess.

The assert is making sure everything that pass here is either an array or an ArrayAccess so both can do $item['somethin'] = $foo;

Reply:
Please switch to the assert lib in this case. our pipeline has an extension installed to get knowledge about that lib. It will give a better error than the native php function.

Reply:
fixed